torch/_inductor/fx_passes/joint_graph.py

```diff
@@ -563,11 +566,50 @@ def pointless_view(match: Match, arg, size):
     """Remove no-op view"""
     node = match.output_node()
     arg_size = list(node.args[0].meta["val"].shape)  # type: ignore[union-attr]
-    if size == arg_size:
+    if guard_size_oblivious(size == arg_size):
```

Comment:
Hmm, I dont think this is sufficient. The individual elements will still be using the guarding version.  Youd want `len(size) == len(arg_size) and all(guard_size_oblivious(dim1, dim2) for zip(size, arg_size))`

Reply:
The api is it takes a symint and returns True if it is True without adding additional guards. Here you are calling `List.__eq__` - that's not going to return a symint. It's going to recursively call the normal, non guard size obvlivious eq on all of its elements.

Reply:
Got it. Do you want me add an `guard_sizes_oblivious` (plural form) API to `symbolic_shapes.py`? Since I need at least call this in two places (in two patterns).

Reply:
SGTM - we could also do pytree recursive guard_size_oblivious impl. but sizes sounds good